wee_slack.py

```diff
@@ -1691,15 +1691,23 @@ def message_from_hash(self, ts_hash, message_filter=None):
             return
         return message
 
-    def message_from_index(self, index, message_filter=None, reverse=True):
-        for ts in (reversed(self.visible_messages) if reverse else self.visible_messages):
-            message = self.messages[ts]
-            if not message_filter or message_filter(message):
-                index -= 1
-                if index == 0:
-                    return message
-
-    def message_from_hash_or_index(self, hash_or_index=None, message_filter=None, reverse=True):
+    def message_from_index(self, index, message_filter=None):
+
+        window = w.current_window()
+        scroll = w.hdata_pointer(hdata.window, window, 'scroll')
+        offset = w.hdata_integer(hdata.window_scroll, scroll, 'lines_after')
```

Comment:
Using the scroll position from the current window makes it behave pretty weird if you have merged buffers. Since lines from multiple buffers are shown, the scroll amount won't match the message offset in one of the buffers.